### PR TITLE
Refactor tests, skip if sklearn not available

### PR DESCRIPTION
**Pull request by krshrimali**
## What does this PR do?

Fixed https://github.com/PyTorchLightning/pytorch-lightning/issues/12058.

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/helpers/datamodules.py [hunk 1]
```diff
@@ -102,6 +101,8 @@ def sample(self):
 
 class ClassifDataModule(SklearnDataModule):
     def __init__(self, num_features=32, length=800, num_classes=3, batch_size=10):
+        if not _SKLEARN_AVAILABLE:
+            pytest.skip("`sklearn` is not available.")
```
Comment (kaushikb11) [inline tests/helpers/datamodules.py:105]: Why is the following condition required here if it's subclassing `SklearnDataModule`?

Comment (carmocca) [reply tests/helpers/datamodules.py:105]: Because it calls `super().__init__` only after the `make_classification()` call which comes from sklearn

**Review by carmocca — APPROVED** [review]

**Review by akihironitta — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]
